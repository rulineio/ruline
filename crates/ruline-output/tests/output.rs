#[cfg(test)]
#[macro_use]
extern crate pretty_assertions;

use dashmap::DashMap;
use insta::assert_snapshot;
use ruline_context::Context;
use ruline_output::Output;
use serde_json::json;

#[test]
fn test_output() {
    let definition = json!({
        "key": {
            "type": "value",
            "value": "value"
        },
        "key2": {
            "type": "data",
            "path": "/my_data"
        },
        "key3": {
            "type": "variable",
            "variable": "var"
        }
    });

    let variables = DashMap::new();
    variables.insert("var".to_string(), json!("val"));
    let context = Context::new(
        json!({
            "my_data": "my_value"
        }),
        variables,
    );

    let output = Output::try_from(definition).unwrap();
    let result = output.process(&context).unwrap();

    assert_eq!(
        result,
        json!({
            "key": "value",
            "key2": "my_value",
            "key3": "val"
        })
    );
}

#[test]
fn test_output_invalid_field_type() {
    let definition = json!({
        "key": {
            "type": "invalid",
            "value": "value"
        }
    });

    let output = Output::try_from(definition);
    assert!(output.is_err());
    assert_snapshot!(output.unwrap_err().to_string());
}
