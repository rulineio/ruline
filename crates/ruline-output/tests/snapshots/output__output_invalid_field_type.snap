---
source: crates/ruline-output/tests/output.rs
expression: output.unwrap_err().to_string()
---
unknown variant `invalid`, expected one of `variable`, `data`, `output`, `value`, `function`
